vpr/src/base/read_place.cpp

```diff
@@ -97,194 +146,113 @@ void read_place(const char* net_file,
 
-    while (ptr != nullptr) {
-        ptr = vtr::strtok(buf, TOKENS, fp, buf);
-        if (ptr == nullptr) {
-            ptr = vtr::fgets(buf, vtr::bufsize, fp);
-            continue; /* Skip blank or comment lines. */
-        }
+    while (std::getline(placement_file, line)) { //Parse line-by-line
+        ++lineno;
 
-        if (strlen(ptr) + 1 < vtr::bufsize) {
-            strcpy(bname, ptr);
-        } else {
-            vpr_throw(VPR_ERROR_PLACE_F, constraints_file, vtr::get_file_line_number_of_last_opened_file(),
-                      "Block name exceeded buffer size of %zu characters", vtr::bufsize);
-        }
+        std::vector<std::string> tokens = vtr::split(line);
 
-        ptr = vtr::strtok(nullptr, TOKENS, fp, buf);
-        if (ptr == nullptr) {
-            vpr_throw(VPR_ERROR_PLACE_F, constraints_file, vtr::get_file_line_number_of_last_opened_file(),
-                      "Incomplete.\n");
-        }
-        sscanf(ptr, "%d", &xtmp);
+        if (tokens.empty()) {
+            continue; //Skip blank lines
 
-        ptr = vtr::strtok(nullptr, TOKENS, fp, buf);
-        if (ptr == nullptr) {
-            vpr_throw(VPR_ERROR_PLACE_F, constraints_file, vtr::get_file_line_number_of_last_opened_file(),
-                      "Incomplete.\n");
-        }
-        sscanf(ptr, "%d", &ytmp);
+        } else if (tokens[0][0] == '#') {
+            continue; //Skip commented lines
 
-        ptr = vtr::strtok(nullptr, TOKENS, fp, buf);
-        if (ptr == nullptr) {
-            vpr_throw(VPR_ERROR_PLACE_F, constraints_file, vtr::get_file_line_number_of_last_opened_file(),
-                      "Incomplete.\n");
-        }
-        int k;
-        sscanf(ptr, "%d", &k);
+        } else if (tokens.size() == 4 || (tokens.size() == 5 && tokens[4][0] == '#')) {
+            //Load the block location
+            //
+            //We should have 4 tokens of actual data, with an optional 5th (commented) token indicating VPR's
+            //internal block number
 
-        ptr = vtr::strtok(nullptr, TOKENS, fp, buf);
-        if (ptr != nullptr) {
-            vpr_throw(VPR_ERROR_PLACE_F, constraints_file, vtr::get_file_line_number_of_last_opened_file(),
-                      "Extra characters at end of line.\n");
-        }
+            std::string block_name = tokens[0];
+            int block_x = vtr::atoi(tokens[1]);
+            int block_y = vtr::atoi(tokens[2]);
+            int sub_tile_index = vtr::atoi(tokens[3]);
 
-        h_ptr = get_hash_entry(hash_table, bname);
-        if (h_ptr == nullptr) {
-            VTR_LOG_WARN("[Line %d] Block %s invalid, no such block.\n",
-                         vtr::get_file_line_number_of_last_opened_file(), bname);
-            ptr = vtr::fgets(buf, vtr::bufsize, fp);
-            continue;
-        }
-        ClusterBlockId bnum(h_ptr->index);
-        int i = xtmp;
-        int j = ytmp;
+            ClusterBlockId blk_id = cluster_ctx.clb_nlist.find_block(block_name);
```

Comment:
find what this returns if it doesn't find the block name - use for invalid block name check